core/config/project_settings.h

```diff
@@ -32,10 +32,6 @@
 #define PROJECT_SETTINGS_H
 
 #include "core/object/class_db.h"
-#include "core/os/thread_safe.h"
-#include "core/templates/hash_map.h"
-#include "core/templates/local_vector.h"
-#include "core/templates/rb_set.h"
 
```

Comment:
```suggestion
#include "core/templates/rb_set.h"

```

Reply:
If this commit passes the gitHub check, then I will not give this include in the header

Reply:
No, that's not valid, it should be in the header regardless

Reply:
I don't understand why? In my opinion, `project_settings.h` is included in 269 files and all of them do not need to include `rb_set.h` (or they themselves have this include). That is, the compiler compiles `rb_set.h` more than __200__ times.

Reply:
They already do, evidently, as it compiles, if they didn't include it *this wouldn't compile*, so if you remove it from the header it won't compile if in the future that include is removed elsewhere

So please this is about basic proper code, don't have a header use classes it does not either include or forward declare, it's unsafe and poor code

Reply:
> They already do, evidently, as it compiles, if they didn't include it _this wouldn't compile_, so if you remove it from the header it won't compile if in the future that include is removed elsewhere
> 
> So please this is about basic proper code, don't have a header use classes it does not either include or forward declare, it's unsafe and poor code
> 
> The simple fact that `project_settings.cpp` doesn't compile without including this proves this
> 
> A header file should compile safely if it is included from an otherwise empty cpp file that just includes the header and any classes forward declared in the header

AAAAAAAAAaaaaaaaaaaaaaaaaaaa, OK. Sounds wise. I'll do it tomorrow.

Reply:
No it's needed because the type is used *in* the class, not just in the cpp